### ISSUE TITLE
aria-hidden=false issues

### ISSUE THREAD

**Issue by stevefaulkner** [w3c/core-aam#58]
The current mapping for [`aria-hidden=false`](https://www.w3.org/TR/core-aam-1.1/#ariaHiddenFalse) is "not mapped" and I mistakenly assumed that _not mapped_ was equal to no effect in this case. Up until  recently  when I became aware of  the chrome behavior whereby if an element has CSS `display:none` set and also has `aria-hidden=false` then the element is exposed to screen reader users and if interactive can be operated. I created a [test case](https://cdpn.io/stevef/debug/jOEMzxw) and filed [a bug on chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=1034303) as its behavior is different from Firefox (which does not expose the element in this case) and I thought that it was [decided by implementors previously](https://bugzilla.mozilla.org/show_bug.cgi?id=945194) that `aria-hidden=false` should have no effect. 
Whatever is decided it would be good to have clarity over what `aria-hidden=false` does and how it should therefore be implemented.

In looking at the particular test case I found that using it on interactive elements in this way (visibly hidden but enabled/available to SR users causes a number of potential issues that map to WCAG criteria:

- [1.3.1 Info and Relationships](1.3.1 Info and Relationships) 
- [2.4.7 Focus Visible ](https://www.w3.org/TR/WCAG21/#focus-visible)

And if so we need to think about whether we want to enshrine WCAG failures in the implementation of this attribute.

**Comment by aleventhal** [comment]
I believe "not mapped" in that table was intended to mean that the property does not have a mapping, e.g. an IA2 state. When something is hidden it gets the invisible state, but when it's visible, there is the absence of a state. I think if aria-hidden="false" was intended to have no effect, then it wouldn't even exist as a possible value.

"Not mapped" is linked to this definition, but I'm not sure if it's 100% clear enough for all of us to be sure:
"There are a number of occurrences in the table where a given state or property is declared "Not mapped". In some cases, this occurs for the default value of the state/property, and is equivalent to its absence. User agents might find it quicker to map the value than check to see if it is the default. For computational efficiency, user agents MAY expose the state or property value if doing so is equivalent to not mapping it. These cases are marked with an asterisk"

**Comment by JAWS-test** [comment]
In my opinion the error is only with Chrome (and not with Firefox or the specification Core AAM), because in the specification it says quite clearly:

> The following elements are not exposed via the accessibility API and user agents MUST NOT include them in the accessibility tree: ...  Elements, including their descendent elements, that have host language semantics specifying that the element is not displayed, such as CSS display:none, visibility:hidden, or the HTML 5 hidden attribute. 
...
If not already excluded from the accessibility tree per the rules above in Excluding Elements in the Accessibility Tree, user agents MUST provide an accessible object in the accessibility tree for DOM elements that meet any of the following criteria:

I.e. there is no way to cancel a hidden being

**Comment by stevefaulkner** [comment]
@JAWS-test wrote:

> but in my opinion not a violation of 2.4.7, because the elements do not get the focus when using the keyboard, but only when reading with the virtual cursor.

So a screen reader users navigating through buttons using the `b` key does not count?

**Comment by JAWS-test** [comment]
> So a screen reader users navigating through buttons using the b key does not count?

As far as I know SC 2.4.7 is about keyboard operation without AT like a screenreader. With the screenreader it is possible to reach invisible content - and in my opinion this is not a violation of 2.4.7. If I use the quick navigation like "b" with a screenreader, the PC cursor is not set to the button, so that no focus is visible even at the visible button.

**Comment by aleventhal** [comment]
I want to make sure Chrome gets this right, but it seems like there are conflicting signals in the specs, and we should clean that up. I agree that the text provided by JAWS-test makes it sound like display:none and visibility:hidden elements should not be exposed.

However, I think the text in aria-hidden overrides that, by saying:
Value   Description
false    The element is exposed to the accessibility API as if it was rendered.

**Comment by JAWS-test** [comment]
> false The element is exposed to the accessibility API as if it was rendered.

This only applies if the area in which aria-hidden is located is not hidden itself (no matter if with aria-hidden or display:none). That aria-hidden=false does not override aria-hidden=true is explicitly stated in the specification. That aria-hidden=false does not override display:none is only implicit in the specification.

Nevertheless, I'm pretty sure it's a bug in Chrome that should be fixed. Whether the specification needs to be formulated more clearly may be decided by others

**Comment by aleventhal** [comment]
> This only applies if the area in which aria-hidden is located is not hidden itself (no matter if with aria-hidden or display:none).

Are you sure that's the intention? I saw somewhere that aria-hidden="true" on an ancestor overrides it, but I don't see anything saying that aria-hidden doesn't override display none.

Not meaning to be argumentative or say that I'm right. I genuinely think this is unclear and that Chrome may be the one that has it right.

What's your opinion on what happens when aria-hidden="false" is on the root of a subtree with display:none (ancestors are all visible)?

**Comment by JAWS-test** [comment]
> Are you sure that's the intention?

This is only my personal interpretation of the specification. Since I am not the author of the specification, I cannot say for sure what the authors were thinking. They have to say that themselves. But I would be very surprised if it would be different, i.e. if a hidden status can be reversed with aria-hidden=false.

**Comment by joanmarie** [comment]
Please note that the content about inclusion is no longer in the Core AAM. The Core AAM is supposed to contain the platform-specific stuff. Whether or not something is included in the accessibility tree should be independent of specific platforms.

https://w3c.github.io/aria/#tree_exclusion
https://w3c.github.io/aria/#tree_inclusion

**Comment by jnurthen** [comment]
related discussion https://github.com/w3c/accname/discussions/116

**Comment by JAWS-test** [comment]
@aleventhal Can you please check the strange behavior of Chrome in relation to aria-hidden=false, see https://github.com/w3c/accname/discussions/116#discussioncomment-504837
